### PR TITLE
Issue 4: Configuration management

### LINKED ISSUE #4: Demonstrate configuration storage
Create a subpackage `github.com/LinuxAtApp/matterleast/spikes/config`
Write some code that looks for `$XDG_CONFIG_HOME/matterleast` and loads some `matterleast.conf` file from inside if it exists. If it doesn't exist, create it.
Importantly, the syntax/contents of the configuration file really doesn't matter. This is about having reference code to load it.

### PR DESCRIPTION
**Pull request by whereswaldon**
Closes #4 
This PR mostly exists for code review purposes. It's not done yet, but feel free to provide feedback on it as it develops. I'd appreciate it.

### CODE REVIEW

**Comment by whereswaldon** [comment]
Now we can load a JSON-formatted config file and access it as a map (a la Java HashMap), so it should be pretty easy to use something like this to store simple configuration. If anyone has time, I'd appreciate a code review. @NaturalFractals @camwood23 @aThorp96

File: spikes/config/main.go [hunk 1]
```diff
@@ -37,4 +38,9 @@ func main() {
 	bytesRead, err := configFile.Read(data)
 	fatal(err)
 
+	config := make(map[string]string)
+	err = json.Unmarshal(data[:bytesRead], &config)
+	fatal(err)
```
Comment (aThorp96) [inline spikes/config/main.go]: This is mostly just coming from my inexperience, but will this allow us to selectively access data? I know we can access any item in the map with the correct key, but how do we get that key, or do we set it using the config file itself?

**Review by aThorp96 — APPROVED** [review]
I like it.

File: spikes/config/main.go (hunk 1 above)
Comment (whereswaldon) [reply spikes/config/main.go]: @aThorp96 We set those keys in the config file. I have no idea what kinds of things they might be right now. I suppose stuff like `server_url`, `username`, and `password`? Anyway, you can iterate over the keys in the map from go to get a list of all of the defined ones using a for loop: `for key, _ := range mapThing {}`

**Pull request merged**